mass2/core/channels.py

```diff
@@ -653,3 +654,57 @@ def from_df(
                 npulses=len(df),
             )
         return Channels(channels, description)
+
+    def save_analysis(self, path: Path | str, overwrite: bool = False) -> None:
+        """Save an analysis-in-progress completely to a directory
+
+        Parameters
+        ----------
+        path : Path | str
+            Directory to save work in. If it doesn't exist, its parent should.
+        overwrite : bool, optional
+            If `path` exists, whether to overwrite it, by default False
+        """
+        if isinstance(path, str):
+            path = pathlib.Path(path)
+        if os.path.exists(path):
+            if os.path.exists(path / "data_all.pkl") and not overwrite:
+                raise ValueError(f"Set save_analysis(...overwrite=True) to overwrite the existing {path=}")
+            if not os.path.isdir(path):
+                raise ValueError(f"save_analysis(path={path}) requires that `path` be a directory, if it exists")
+        else:
+            os.mkdir(path)
+            assert os.path.exists(path) and os.path.isdir(path)
+
+        channels = {}
+        for ch_num, ch in self.channels.items():
+            fname = path / f"data_chan{ch_num:04d}.parquet"
+            df = ch.df.drop("pulse", "timestamp", "subframecount")
+            df.write_parquet(fname)
+            steps = ch.steps.trim_dead_ends(None, drop_debug=True)
```

Comment:
We should should not `trim_dead_ends` here. Ohh I see None preserves all columns. Still.... why bother dropping debug? This is an analysis in progress, I'd rather have debug info.

Reply:
My reasoning was that the debug info was "fat". But I don't know quantitatively what this accomplishes. Maybe we should skip it. I certainly see why it would be a nice-to-have after reloading.